toogoodtogo_ha_mqtt_bridge/main.py

```diff
@@ -199,21 +199,70 @@ def check_for_removed_stores(shops: []):
 def loop(event):
     logger.info("Starting loop")
 
+    create_data_dir()
     token_exits = check_existing_token_file()
     tgtg_client.login()
     if not token_exits and tgtg_client.access_token:
         write_token_file()
 
     while True:
         logger.debug("Loop run started")
-        if not check():
-            logger.error("Loop was not successfully.")
+        if check_pollingtime():
+            if not check():
+                logger.error("Loop was not successfully.")
+            else:
+                logger.debug("Loop run finished")
+                watchdog.reset()
         else:
-            logger.debug("Loop run finished")
-            watchdog.reset()
+            send_waiting_state()
+            logger.debug("No polling defined for the current time. Loop run finished.")
         event.wait(settings.tgtg.every_n_minutes * 60)
 
 
+def create_data_dir():
+    data_dir = settings.get("data_dir")
+    if not os.path.isdir(data_dir):
+        os.mkdir(data_dir)
+
+
+def check_pollingtime():
+    splitted_start = settings.get("poll_from").split(":")
+    splitted_end = settings.get("poll_until").split(":")
+    start_time = time(int(splitted_start[0]), int(splitted_start[1]))
+    end_time = time(int(splitted_end[0]), int(splitted_end[1]))
+    now = datetime.now().time()
+
+    if start_time <= now <= end_time:
```

Comment:
This came with some corner cases that should be taken care of. 

E.g. when you set `end_time` after midnight it never returns `True`:

![image](https://user-images.githubusercontent.com/5927148/144728311-b04d97a6-3455-4be2-8542-83850040da51.png)

Reply:
Some small additional if might solve this:
https://stackoverflow.com/a/10748028/635876